code/general/loop-vs-while-true-vs-until-false-vs-infinite-range.rb

```diff
@@ -0,0 +1,42 @@
+require "benchmark/ips"
+
+NUMBER = 100_000_000
+
+def fastest
+  index = 0
+  while true
+    break if index > NUMBER
+    index += 1
+  end
+end
+
+def fast
+  index = 0
+  until false
+    break if index > NUMBER
+    index += 1
+  end
+end
+
+def slow
+  index = 0
+  loop do
+    break if index > NUMBER
+    index += 1
+  end
+end
+
+def slowest
+  (0..).each do |index|
+    break if index > NUMBER
+    index += 1
```

Comment:
```suggestion
```

The index is already incremeted (you break too early at `Number / 2`)!

Reply:
Oops. I've fixed it. Thanks.